pandas/io/common.py

```diff
@@ -244,6 +272,106 @@ def file_path_to_url(path):
+        //<user>:<password>@<host>:<port>/<url-path>
+
+        .. versionadded:: 0.21.0
+
+    Returns
+    -------
+    (username, password), url_no_usrpwd : tuple, string  Default ('', '') url
+        A tuple with (username, pwd) pair and
+        url without username or password (if it contained it )
+
+    Raises
+    ------
+    ValueError for empty url
+    """
+    if not url_with_uname:
+        msg = "Empty url: {_type}"
+        raise ValueError(msg.format(_type=type(url_with_uname)))
+    o = parse_url(url_with_uname)
+    uname = o.username if o.username else ''
+    pwd = o.password if o.password else ''
+    url_no_usrpwd = url_with_uname
+    if uname or pwd:
+        usrch = '{}:{}@{}'.format(o.username, o.password, o.hostname)
+        url_no_usrpwd = url_with_uname.replace(usrch, o.hostname)
+    return (uname, pwd), url_no_usrpwd
+
+
+def get_urlopen_args(url_with_uname, auth=None, verify_ssl=True):
+    """
+    generate args to pass to urlopen - including basic auth and and support
+    for disabling verification of SSL certificates ( useful where
+    self-signed SSL certificates are acceptable security risk -eg: Testing )
+
+    Parameters
+    ----------
+    url_with_uname : string
+        a url that may or may not contain username and password
+        see section 3.1 RFC 1738 https://www.ietf.org/rfc/rfc1738.txt
+        //<user>:<password>@<host>:<port>/<url-path>
+
+        .. versionadded:: 0.21.0
+
+    auth : tuple, default None
+        A tuple of string with (username, password) string for
+        HTTP(s) basic auth: eg auth= ('roberto', 'panda$4life')
+
+        .. versionadded:: 0.21.0
+
+    verify_ssl : boolean, Default True
+        If False, allow self signed and invalid SSL certificates for https
+
+        .. versionadded:: 0.21.0
+
+    Returns
+    -------
+    Request, kwargs to pass to urlopen. kwargs may be {} or {'context': obj }
+
+    Raises
+    ------
+    ValueError if only one of username or password is provided.
+    """
+    uname = pwd = None
+    url_no_usrpwd = url_with_uname
+    if auth and len(auth) == 2:
+        uname, pwd = auth
+    if not uname and not pwd:
+        (uname, pwd), url_no_usrpwd = split_auth_from_url(url_with_uname)
+    req = Request(url_no_usrpwd)
+    if uname and pwd:
+        upstr = '{}:{}'.format(uname, pwd)
+        if compat.PY3:
+            b64str = base64.b64encode(bytes(upstr, 'ascii')).decode('utf-8')
+        else:
+            b64str = base64.encodestring(upstr).replace('\n', '')
+        req.add_header("Authorization", "Basic {}".format(b64str))
+    elif uname or pwd:
+        raise ValueError('Only username or password provided without providing the other')
+    kwargs = {}
+    if verify_ssl not in [None, True]:
+        kwargs['context'] = ssl._create_unverified_context()
+        msg = 'SSL certificate verification is being disabled for HTTPS. Possible security risk'
+        warnings.warn(msg, InsecureRequestWarning)
```

Comment:
For this, you can use the `assert_produces_warning` context that we have defined in `pandas.util.testing`.